src/image/cache/minio-s3/index.ts

```diff
@@ -0,0 +1,60 @@
+import ImageCache from '../abstract'
+import { createHash } from 'crypto'
+import { Client } from 'minio'
+
+export default class MinioS3ImageCache extends ImageCache {
+  private static client: Client
+
+  public constructor (config, req) {
+    super(config, req)
+    if (MinioS3ImageCache.client === undefined) {
+      MinioS3ImageCache.client = new Client(this.moduleConfig.libraryOptions)
+    }
+  }
+
+  public get moduleConfig (): any {
+    return this.config.imageable.caching[`minio-s3`]
+  }
+
+  public async getImageFromCache () {
+    await new Promise((resolve, reject) => {
+      const chunks = []
+      MinioS3ImageCache.client.getObject(this.moduleConfig.bucket, this.key, (err, stream) => {
+        if (err) reject(err)
+        stream.on('data', (chunk) => {
+          chunks.push(chunk)
+        })
+        stream.on('end', () => {
+          this.image = Buffer.concat(chunks)
+          resolve()
+        })
+        stream.on('error', (err) => {
+          reject(err)
+        })
+      })
+    }).catch((e) => {
+      console.log(e)
+    })
+  }
+
+  public async save () {
+    await MinioS3ImageCache.client.putObject(this.moduleConfig.bucket, this.key, this.image)
+  }
+
+  public async check () {
+    try {
+      await MinioS3ImageCache.client.statObject(this.moduleConfig.bucket, this.key)
+      return true
+    } catch (e) {
+      return false
```

Comment:
Maybe it would be worth to print this possible error - at least in `console`? It could be hard to debug if we hide it

Reply:
Problem is: minio-api has no 'exists'-method...so we need to stat the file, which actually returns an error, if non-existent. I know it is far from optimal to use this for control flow - maybe you have a better solution? See: https://github.com/minio/minio-js/pull/714#issuecomment-417959840

Reply:
Understood. Now, I agree with your solution. Do you @gibkigonzo ?